<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Pseudotime-Aware MI Network Analysis with miEdgeR • miEdgeR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Pseudotime-Aware MI Network Analysis with miEdgeR">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">miEdgeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/index.html">Articles</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav"></ul>
</div>


  </div>
</nav><div class="container template-article">



<link href="pseudotime_mi_network_files/tabwid-1.1.3/tabwid.css" rel="stylesheet">
<script src="pseudotime_mi_network_files/tabwid-1.1.3/tabwid.js"></script><div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Pseudotime-Aware MI Network Analysis with miEdgeR</h1>
            
      

      <div class="d-none name"><code>pseudotime_mi_network.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="about-this-vignette">About this vignette<a class="anchor" aria-label="anchor" href="#about-this-vignette"></a>
</h2>
<p>This tutorial demonstrates how to incorporate pseudotime information
into mutual-information (MI) network analysis using miEdgeR. We bin
cells along a trajectory, build MI networks per bin, detect communities,
and track dynamic changes in modules, hub genes, and enrichment across
pseudotime.</p>
</div>
<div class="section level2">
<h2 id="prerequisites">Prerequisites:<a class="anchor" aria-label="anchor" href="#prerequisites"></a>
</h2>
<p>R (≥ 4.2.0)</p>
<p>Seurat object containing cluster metadata (RNA_snn_res.0.2) and a
numeric pseudotime column</p>
<p>Packages: miEdgeR, Seurat, igraph, clusterProfiler, org.Hs.eg.db,
ggplot2, ggraph, flextable, pheatmap</p>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>We extend the MI-network workflow by leveraging single-cell
pseudotime. After generating a pseudotime vector, we bin cells into
discrete stages, compute MI networks in each bin, and then analyze how
community structure, hub genes, and functional enrichment evolve along
the trajectory.</p>
</div>
<div class="section level2">
<h2 id="load-libraries-and-data">Load Libraries and Data<a class="anchor" aria-label="anchor" href="#load-libraries-and-data"></a>
</h2>
<p>Load all dependencies and read the preprocessed Seurat object. Ensure
your object includes clustering and pseudotime metadata.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://satijalab.org/seurat" class="external-link">Seurat</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://iichelhadi.github.io/miEdgeR/">miEdgeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igraph.org" class="external-link">igraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">clusterProfiler</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">org.Hs.eg.db</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggraph.data-imaginist.com" class="external-link">ggraph</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ardata-fr.github.io/flextable-book/" class="external-link">flextable</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">pheatmap</span><span class="op">)</span></span>
<span></span>
<span><span class="va">seurat_obj</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="st">'pbmc3k.rds'</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="generate-or-verify-pseudotime">Generate or Verify Pseudotime<a class="anchor" aria-label="anchor" href="#generate-or-verify-pseudotime"></a>
</h2>
<p>Here we generate a <strong>mock</strong> pseudotime for cluster 1 and
plot its distribution.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span><span class="va">cluster_1_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">[</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">RNA_snn_res.0.1</span> <span class="op">==</span> <span class="st">"1"</span><span class="op">]</span></span>
<span><span class="va">mock_pseudotime</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/numeric.html" class="external-link">numeric</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">ncol</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mock_pseudotime</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span></span>
<span><span class="va">mock_pseudotime</span><span class="op">[</span><span class="va">cluster_1_cells</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq</a></span><span class="op">(</span><span class="fl">0</span>, <span class="fl">1</span>, length.out <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cluster_1_cells</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">other_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span>, <span class="va">cluster_1_cells</span><span class="op">)</span></span>
<span><span class="va">mock_pseudotime</span><span class="op">[</span><span class="va">other_cells</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Uniform.html" class="external-link">runif</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">other_cells</span><span class="op">)</span>, <span class="fl">0</span>, <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">pseudotime</span> <span class="op">&lt;-</span> <span class="va">mock_pseudotime</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/match.html" class="external-link">match</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">mock_pseudotime</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">p_pt</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>pseudotime <span class="op">=</span> <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span><span class="va">cluster_1_cells</span>, <span class="st">"pseudotime"</span><span class="op">]</span><span class="op">)</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">pseudotime</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">30</span>, fill <span class="op">=</span> <span class="st">"#f56642"</span>, color <span class="op">=</span> <span class="st">"black"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Pseudotime Distribution for Cluster 1"</span>, x <span class="op">=</span> <span class="st">"Pseudotime"</span>, y <span class="op">=</span> <span class="st">"Cell Count"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_pt</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-2-1.png" alt="pseudotime distribution histogram" width="700"></p>
<!-- ## Verify Pseudotime -->
<!-- Confirm pseudotime data for selected cluster. -->
<!-- ```{r} -->
<!-- seurat_obj@meta.data$pseudotime <- seurat_obj@meta.data$Lineage2_2 -->
<!-- cluster_1_cells <- colnames(seurat_obj)[seurat_obj@meta.data$monocle_clusters2 == "5"] -->
<!-- cat("Cluster 5 cells:", length(cluster_1_cells), "\n") -->
<!-- cat("Pseudotime non-NAs:", sum(!is.na(seurat_obj@meta.data[cluster_1_cells, "pseudotime"])), "\n") -->
<!-- cat("Pseudotime range:", range(seurat_obj@meta.data[cluster_1_cells, "pseudotime"], na.rm = TRUE), "\n") -->
<!-- p_pt <- ggplot(data.frame(pseudotime = seurat_obj@meta.data[cluster_1_cells, "pseudotime"]), aes(x = pseudotime)) + -->
<!--   geom_histogram(bins = 30, fill = "#f56642", color = "black") + -->
<!--   labs(title = "Pseudotime Distribution for Cluster 5", x = "Pseudotime", y = "Cell Count") + -->
<!--   theme_minimal() -->
<!-- print(p_pt) -->
<!-- ``` -->
</div>
<div class="section level2">
<h2 id="compute-pseudotime-mi-networks">Compute Pseudotime MI Networks<a class="anchor" aria-label="anchor" href="#compute-pseudotime-mi-networks"></a>
</h2>
<p>For each pseudotime bin <strong>(3 bins)</strong>, we extract cells,
filter top variable genes, compute MI in parallel, and build thresholded
igraph objects.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pt_results</span> <span class="op">&lt;-</span> <span class="fu">miEdgeR</span><span class="fu">::</span><span class="fu"><a href="../reference/compute_pseudotime_mi_network.html">compute_pseudotime_mi_network</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span>,</span>
<span>  cluster_id <span class="op">=</span> <span class="st">"1"</span>,</span>
<span>  cluster_field <span class="op">=</span> <span class="st">"RNA_snn_res.0.1"</span>,</span>
<span>  pseudotime_col <span class="op">=</span> <span class="st">"pseudotime"</span>,</span>
<span>  assay_name <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  counts_layer <span class="op">=</span> <span class="st">"counts"</span>,</span>
<span>  data_layer <span class="op">=</span> <span class="st">"data"</span>,</span>
<span>  min_expr_pct <span class="op">=</span> <span class="fl">0.05</span>,</span>
<span>  top_n_genes <span class="op">=</span> <span class="fl">3000</span>,</span>
<span>  min_cells_bin <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  n_bins <span class="op">=</span> <span class="fl">3</span>, <span class="co">## set the number of pseudotime bins</span></span>
<span>  n_cores <span class="op">=</span> <span class="fl">65</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Using adaptive nbins = 5 </span></span>
<span><span class="co">## Using adaptive nbins = 5 </span></span>
<span><span class="co">## Using adaptive nbins = 5</span></span></code></pre>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Valid bins:"</span>, <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">sapply</a></span><span class="op">(</span><span class="va">pt_results</span>, <span class="va">is.null</span><span class="op">)</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Valid bins: 3</span></span></code></pre>
</div>
<div class="section level2">
<h2 id="summarize-bin-graphs">Summarize Bin Graphs<a class="anchor" aria-label="anchor" href="#summarize-bin-graphs"></a>
</h2>
<p>Quick summary of node and edge counts and density per bin to assess
network complexity over pseudotime.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">for</span> <span class="op">(</span><span class="va">bin</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">[[</span><span class="va">bin</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Bin"</span>, <span class="va">bin</span>, <span class="st">"- Vertices:"</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/gorder.html" class="external-link">vcount</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">[[</span><span class="va">bin</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Bin"</span>, <span class="va">bin</span>, <span class="st">"- Edges:"</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/gsize.html" class="external-link">ecount</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">[[</span><span class="va">bin</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Bin"</span>, <span class="va">bin</span>, <span class="st">"- Density:"</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/edge_density.html" class="external-link">graph.density</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">[[</span><span class="va">bin</span><span class="op">]</span><span class="op">]</span><span class="op">)</span>, <span class="st">"\n"</span><span class="op">)</span></span>
<span>  <span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Bin"</span>, <span class="va">bin</span>, <span class="st">"- Skipped (insufficient cells)\n"</span><span class="op">)</span></span>
<span>  <span class="op">}</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">## Bin PT1 - Vertices: 2911 </span></span>
<span><span class="co">## Bin PT1 - Edges: 211806 </span></span>
<span><span class="co">## Bin PT1 - Density: 0.05000726 </span></span>
<span><span class="co">## Bin PT2 - Vertices: 2911 </span></span>
<span><span class="co">## Bin PT2 - Edges: 211825 </span></span>
<span><span class="co">## Bin PT2 - Density: 0.05001175 </span></span>
<span><span class="co">## Bin PT3 - Vertices: 2911 </span></span>
<span><span class="co">## Bin PT3 - Edges: 211818 </span></span>
<span><span class="co">## Bin PT3 - Density: 0.05001009</span></span></code></pre>
<p>All bins yielded valid networks with consistent structure (2911
genes, ~211K edges, density ≈ 0.05), indicating stable co-regulatory
patterns across pseudotime stages in Cluster 1.</p>
</div>
<div class="section level2">
<h2 id="community-detection">Community Detection<a class="anchor" aria-label="anchor" href="#community-detection"></a>
</h2>
<p>Detect robust gene modules per bin by percolation (noise filtering)
followed by Leiden clustering and neighborhood-overlap expansion.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comm_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/detect_communities_across_bins.html">detect_communities_across_bins</a></span><span class="op">(</span></span>
<span>  <span class="va">pt_results</span>,</span>
<span>  min_size <span class="op">=</span> <span class="fl">50</span>,</span>
<span>  n_iterations <span class="op">=</span> <span class="fl">100</span>,</span>
<span>  edge_fraction <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>  freq_threshold <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  percentile <span class="op">=</span> <span class="fl">0.95</span>,</span>
<span>  resolution_parameter <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  overlap_threshold <span class="op">=</span> <span class="fl">0.1</span></span>
<span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Bin PT1 - Communities (size &gt;= 50 ): 1 </span></span>
<span><span class="co">## Sizes: 1017 </span></span>
<span><span class="co">## Bin PT2 - Communities (size &gt;= 50 ): 2 </span></span>
<span><span class="co">## Sizes: 957 590 </span></span>
<span><span class="co">## Bin PT3 - Communities (size &gt;= 50 ): 2 </span></span>
<span><span class="co">## Sizes: 970 694</span></span></code></pre>
<p>We applied <code><a href="../reference/detect_communities_across_bins.html">detect_communities_across_bins()</a></code> to identify
stable gene communities across pseudotime bins. Each bin’s consensus MI
network revealed large co-expression modules, with Bin PT1 showing one
major community (size: 1017), while Bins PT2 and PT3 displayed two
communities each (sizes: 957/590 and 970/694, respectively). This
suggests dynamic reorganization of gene modules during pseudotime
progression in Cluster 1.</p>
</div>
<div class="section level2">
<h2 id="community-sizes">Community Sizes<a class="anchor" aria-label="anchor" href="#community-sizes"></a>
</h2>
<p>Bar plot showing how module sizes vary across pseudotime bins.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">comm_sizes_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">comm_results</span>, <span class="kw">function</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Bin <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">bin</span>, Community <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">sizes</span><span class="op">)</span><span class="op">)</span>, Size <span class="op">=</span> <span class="va">res</span><span class="op">$</span><span class="va">sizes</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_comm_sizes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">comm_sizes_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Bin</span>, y <span class="op">=</span> <span class="va">Size</span>, fill <span class="op">=</span> <span class="va">Community</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"dodge"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Community Sizes Across Pseudotime Bins (C5)"</span>, x <span class="op">=</span> <span class="st">"Pseudotime Bin"</span>, y <span class="op">=</span> <span class="st">"Community Size"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_comm_sizes</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-6-1.png" alt="bar plot of module sizes per bin" width="672"></p>
</div>
<div class="section level2">
<h2 id="go-enrichment">GO Enrichment<a class="anchor" aria-label="anchor" href="#go-enrichment"></a>
</h2>
<p>Perform enrichment for communities after filtering of housekeeping
genes which tend to be over-represented due to their high mi-scores</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://yulab-smu.top/biomedical-knowledge-mining-book/" class="external-link">DOSE</a></span><span class="op">)</span></span>
<span><span class="va">enrich_results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/enrich_go.html">enrich_go</a></span><span class="op">(</span><span class="va">comm_results</span>,semantic_space <span class="op">=</span> <span class="st">'GO'</span>,ontology <span class="op">=</span> <span class="st">'BP'</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Community 1 in bin PT1 - Genes after filtering: 1017 </span></span>
<span><span class="co">## Community 1 in bin PT2 - Genes after filtering: 957 </span></span>
<span><span class="co">## Community 2 in bin PT2 - Genes after filtering: 590 </span></span>
<span><span class="co">## Community 1 in bin PT3 - Genes after filtering: 970 </span></span>
<span><span class="co">## Community 2 in bin PT3 - Genes after filtering: 694 </span></span>
<span><span class="co">## Enrichment for PT1_Group_1 :</span></span>
<span><span class="co">##                    ID                                            Description</span></span>
<span><span class="co">## GO:0022904 GO:0022904                   respiratory electron transport chain</span></span>
<span><span class="co">## GO:0042773 GO:0042773               ATP synthesis coupled electron transport</span></span>
<span><span class="co">## GO:0042775 GO:0042775 mitochondrial ATP synthesis coupled electron transport</span></span>
<span><span class="co">## GO:0046034 GO:0046034                                  ATP metabolic process</span></span>
<span><span class="co">##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0022904    45/891 113/18800 3.031250e-30 7.886921e-27 5.938203e-27</span></span>
<span><span class="co">## GO:0042773    41/891  92/18800 4.949950e-30 7.886921e-27 5.938203e-27</span></span>
<span><span class="co">## GO:0042775    41/891  92/18800 4.949950e-30 7.886921e-27 5.938203e-27</span></span>
<span><span class="co">## GO:0046034    59/891 273/18800 3.464913e-23 1.656228e-20 1.247004e-20</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                                             geneID</span></span>
<span><span class="co">## GO:0022904                                                                                    BID/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2L/COX7C/DGUOK/GHITM/ISCU/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB5/NDUFB7/NDUFB9/NDUFS5/NDUFS7/NDUFV2/PARK7/SDHB/SOD2/UQCR10/UQCRFS1/UQCRH/NDUFB4/NDUFC2/NDUFS6/CYCS/NDUFA3/SDHD/UQCRQ/ETFB/NDUFS8/NDUFA6/DNAJC15/NDUFS3/SCO2/NDUFA2/NDUFS2/SDHC/NDUFAB1</span></span>
<span><span class="co">## GO:0042773                                                                                                        BID/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2L/COX7C/DGUOK/GHITM/ISCU/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB5/NDUFB7/NDUFB9/NDUFS5/NDUFS7/NDUFV2/PARK7/UQCR10/UQCRFS1/UQCRH/NDUFB4/NDUFC2/NDUFS6/CYCS/NDUFA3/SDHD/UQCRQ/NDUFS8/NDUFA6/DNAJC15/NDUFS3/NDUFA2/NDUFS2/SDHC/NDUFAB1</span></span>
<span><span class="co">## GO:0042775                                                                                                        BID/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2L/COX7C/DGUOK/GHITM/ISCU/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB5/NDUFB7/NDUFB9/NDUFS5/NDUFS7/NDUFV2/PARK7/UQCR10/UQCRFS1/UQCRH/NDUFB4/NDUFC2/NDUFS6/CYCS/NDUFA3/SDHD/UQCRQ/NDUFS8/NDUFA6/DNAJC15/NDUFS3/NDUFA2/NDUFS2/SDHC/NDUFAB1</span></span>
<span><span class="co">## GO:0046034 BID/CHCHD10/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2/COX7A2L/COX7C/DGUOK/EIF6/ENO1/FIS1/GHITM/HSPA8/ISCU/LDHA/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB5/NDUFB7/NDUFB9/NDUFS5/NDUFS7/NDUFV2/PARK7/PGAM1/PGK1/PKM/RHOA/TGFB1/TPI1/TSPO/UQCR10/UQCRC1/UQCRFS1/UQCRH/NDUFB4/NDUFC2/NDUFS6/CYCS/NDUFA3/SDHD/UQCRQ/TREX1/UQCRC2/NDUFS8/NDUFA6/DNAJC15/NDUFS3/NDUFA2/NDUFS2/ALDOA/SDHC/NDUFAB1</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0022904    45</span></span>
<span><span class="co">## GO:0042773    41</span></span>
<span><span class="co">## GO:0042775    41</span></span>
<span><span class="co">## GO:0046034    59</span></span>
<span><span class="co">## Enrichment for PT2_Group_1 :</span></span>
<span><span class="co">##                    ID                          Description GeneRatio   BgRatio</span></span>
<span><span class="co">## GO:0022904 GO:0022904 respiratory electron transport chain    41/832 113/18800</span></span>
<span><span class="co">## GO:0019882 GO:0019882  antigen processing and presentation    36/832 107/18800</span></span>
<span><span class="co">## GO:0046034 GO:0046034                ATP metabolic process    56/832 273/18800</span></span>
<span><span class="co">## GO:0002396 GO:0002396         MHC protein complex assembly    14/832  20/18800</span></span>
<span><span class="co">##                  pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0022904 6.164300e-27 2.928659e-23 2.186704e-23</span></span>
<span><span class="co">## GO:0019882 1.673296e-22 8.833145e-20 6.595331e-20</span></span>
<span><span class="co">## GO:0046034 2.487528e-22 1.181825e-19 8.824180e-20</span></span>
<span><span class="co">## GO:0002396 3.007261e-15 7.143747e-13 5.333931e-13</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                      geneID</span></span>
<span><span class="co">## GO:0022904                                                                                          BID/CHCHD2/COX5A/COX5B/COX6A1/COX7A2L/COX7C/DGUOK/GHITM/ISCU/NDUFA2/NDUFA4/NDUFB10/NDUFB2/NDUFB5/NDUFB7/NDUFB9/NDUFS3/NDUFS5/NDUFS7/NDUFV2/PARK7/SDHB/SDHD/SOD2/UQCR10/UQCRFS1/UQCRH/UQCRQ/NDUFS8/NDUFA1/NDUFB4/NDUFA3/COX4I1/SCO2/NDUFA12/ETFB/CYC1/NDUFC2/SDHC/NDUFS6</span></span>
<span><span class="co">## GO:0019882                                                                                                                      CALR/CD68/CD74/CTSD/CTSH/HLA-A/HLA-B/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/HLA-E/HLA-F/IFI30/PDIA3/PSMB8/PSME1/PYCARD/TAPBP/WAS/FGL2/HLA-DMB/HLA-DQA2/SLC11A1/RAB10/CTSS/FCER1G/HLA-C/LILRB2/B2M/TREX1/CTSL</span></span>
<span><span class="co">## GO:0046034 ALDOA/BID/CHCHD2/COX5A/COX5B/COX6A1/COX7A2/COX7A2L/COX7C/DGUOK/ENO1/FIS1/GHITM/HSPA8/ISCU/LDHA/NDUFA2/NDUFA4/NDUFB10/NDUFB2/NDUFB5/NDUFB7/NDUFB9/NDUFS3/NDUFS5/NDUFS7/NDUFV2/PARK7/PGAM1/PGK1/PKM/RHOA/SDHD/TGFB1/TPI1/UQCR10/UQCRC1/UQCRFS1/UQCRH/UQCRQ/CHCHD10/EIF6/NDUFS8/NDUFA1/NDUFB4/NDUFA3/NCOR1/COX4I1/NDUFA12/UQCRC2/CYC1/NDUFC2/SDHC/NDUFS6/TSPO/TREX1</span></span>
<span><span class="co">## GO:0002396                                                                                                                                                                                                                                                      CALR/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/PDIA3/TAPBP/HLA-DMB/HLA-DQA2/B2M</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0022904    41</span></span>
<span><span class="co">## GO:0019882    36</span></span>
<span><span class="co">## GO:0046034    56</span></span>
<span><span class="co">## GO:0002396    14</span></span>
<span><span class="co">## Enrichment for PT2_Group_2 :</span></span>
<span><span class="co">##                    ID                         Description GeneRatio   BgRatio</span></span>
<span><span class="co">## GO:0019882 GO:0019882 antigen processing and presentation    33/525 107/18800</span></span>
<span><span class="co">## GO:0045333 GO:0045333                cellular respiration    35/525 231/18800</span></span>
<span><span class="co">## GO:0050863 GO:0050863     regulation of T cell activation    40/525 342/18800</span></span>
<span><span class="co">## GO:0002396 GO:0002396        MHC protein complex assembly    12/525  20/18800</span></span>
<span><span class="co">##                  pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0019882 1.217336e-25 4.959429e-22 3.851909e-22</span></span>
<span><span class="co">## GO:0045333 2.873605e-16 1.463383e-13 1.136586e-13</span></span>
<span><span class="co">## GO:0050863 1.763186e-14 6.400725e-12 4.971341e-12</span></span>
<span><span class="co">## GO:0002396 2.042450e-14 6.400725e-12 4.971341e-12</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                  geneID</span></span>
<span><span class="co">## GO:0019882                                                  CTSS/FCER1G/HLA-C/LILRB2/B2M/TREX1/CTSL/CD74/CTSD/CTSH/HLA-A/HLA-B/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/HLA-E/HLA-F/IFI30/PDIA3/PSMB8/PSME1/PYCARD/WAS/FGL2/HLA-DMB/HLA-DQA2/SLC11A1/RAB10</span></span>
<span><span class="co">## GO:0045333                                                  TREX1/MDH2/MDH1/IDH2/ISCU/NDUFB10/NDUFB2/NDUFB5/NDUFB9/OXA1L/PARK7/PRELID1/RHOA/SDHB/SDHD/SOD2/UQCR10/UQCRC1/UQCRFS1/UQCRH/UQCRQ/CHCHD10/CAT/NDUFS8/NDUFA1/NDUFB4/COX4I1/SUCLG1/SCO2/NDUFA12/UQCRC2/CYC1/NDUFC2/SDHC/NDUFS6</span></span>
<span><span class="co">## GO:0050863 AIF1/LILRB1/LILRB2/SPN/B2M/LGALS1/TREX1/ADA/RUNX3/CD74/CEBPB/CORO1A/HLA-A/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/HLA-E/LGALS3/LGALS9/LYN/NFKBIZ/PRELID1/PTPN6/PTPRC/PYCARD/RAC2/RHOA/SOD1/TNFRSF1B/TNFSF13B/IRF1/FGL2/CD300A/HLA-DMB/HLA-DQA2</span></span>
<span><span class="co">## GO:0002396                                                                                                                                                                             B2M/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/PDIA3/HLA-DMB/HLA-DQA2</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0019882    33</span></span>
<span><span class="co">## GO:0045333    35</span></span>
<span><span class="co">## GO:0050863    40</span></span>
<span><span class="co">## GO:0002396    12</span></span>
<span><span class="co">## Enrichment for PT3_Group_1 :</span></span>
<span><span class="co">##                    ID                                            Description</span></span>
<span><span class="co">## GO:0042773 GO:0042773               ATP synthesis coupled electron transport</span></span>
<span><span class="co">## GO:0042775 GO:0042775 mitochondrial ATP synthesis coupled electron transport</span></span>
<span><span class="co">## GO:0019882 GO:0019882                    antigen processing and presentation</span></span>
<span><span class="co">## GO:0046034 GO:0046034                                  ATP metabolic process</span></span>
<span><span class="co">##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0042773    39/847  92/18800 1.962503e-28 4.733558e-25 3.503585e-25</span></span>
<span><span class="co">## GO:0042775    39/847  92/18800 1.962503e-28 4.733558e-25 3.503585e-25</span></span>
<span><span class="co">## GO:0019882    36/847 107/18800 3.047327e-22 1.633367e-19 1.208951e-19</span></span>
<span><span class="co">## GO:0046034    56/847 273/18800 5.858412e-22 2.826098e-19 2.091762e-19</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                                                                                                                       geneID</span></span>
<span><span class="co">## GO:0042773                                                                                                   BID/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2L/COX7C/CYC1/CYCS/ISCU/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB4/NDUFB9/NDUFC2/NDUFS5/NDUFS7/PARK7/UQCR10/UQCRFS1/UQCRH/NDUFA3/NDUFAB1/NDUFS6/SDHC/NDUFV2/UQCRQ/DGUOK/GHITM/NDUFB7/SDHD/NDUFA2/NDUFS8/NDUFB5/NDUFS2</span></span>
<span><span class="co">## GO:0042775                                                                                                   BID/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2L/COX7C/CYC1/CYCS/ISCU/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB4/NDUFB9/NDUFC2/NDUFS5/NDUFS7/PARK7/UQCR10/UQCRFS1/UQCRH/NDUFA3/NDUFAB1/NDUFS6/SDHC/NDUFV2/UQCRQ/DGUOK/GHITM/NDUFB7/SDHD/NDUFA2/NDUFS8/NDUFB5/NDUFS2</span></span>
<span><span class="co">## GO:0019882                                                                                                                        CD68/CD74/CTSD/CTSH/FGL2/HLA-A/HLA-C/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQA2/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/HLA-E/HLA-F/IFI30/LILRB2/PSMB8/PSME1/PYCARD/RAB10/RAB32/SLC11A1/WAS/HLA-B/HLA-DMB/CALR/PDIA3/TAP1/CTSS/FCER1G/B2M/CTSL</span></span>
<span><span class="co">## GO:0046034 ALDOA/BID/CHCHD10/CHCHD2/COX4I1/COX5A/COX5B/COX6A1/COX7A2/COX7A2L/COX7C/CYC1/CYCS/ENO1/FIS1/HSPA8/ISCU/LDHA/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB4/NDUFB9/NDUFC2/NDUFS5/NDUFS7/PARK7/PGAM1/PGK1/PKM/RHOA/TPI1/UQCR10/UQCRC2/UQCRFS1/UQCRH/NDUFA3/NDUFAB1/TGFB1/NDUFS6/SDHC/NDUFV2/UQCRQ/DGUOK/GHITM/NDUFB7/SDHD/UQCRC1/NCOR1/NDUFA2/NDUFS8/NDUFB5/NDUFS2/TSPO</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0042773    39</span></span>
<span><span class="co">## GO:0042775    39</span></span>
<span><span class="co">## GO:0019882    36</span></span>
<span><span class="co">## GO:0046034    56</span></span>
<span><span class="co">## Enrichment for PT3_Group_2 :</span></span>
<span><span class="co">##                    ID                                      Description</span></span>
<span><span class="co">## GO:0019882 GO:0019882              antigen processing and presentation</span></span>
<span><span class="co">## GO:0022904 GO:0022904             respiratory electron transport chain</span></span>
<span><span class="co">## GO:0046034 GO:0046034                            ATP metabolic process</span></span>
<span><span class="co">## GO:0033108 GO:0033108 mitochondrial respiratory chain complex assembly</span></span>
<span><span class="co">##            GeneRatio   BgRatio       pvalue     p.adjust       qvalue</span></span>
<span><span class="co">## GO:0019882    34/616 107/18800 1.354942e-24 6.066077e-21 4.622493e-21</span></span>
<span><span class="co">## GO:0022904    32/616 113/18800 1.720275e-21 2.567224e-18 1.956285e-18</span></span>
<span><span class="co">## GO:0046034    42/616 273/18800 5.208031e-17 1.457272e-14 1.110475e-14</span></span>
<span><span class="co">## GO:0033108    25/616  94/18800 2.341378e-16 5.823528e-14 4.437664e-14</span></span>
<span><span class="co">##                                                                                                                                                                                                                                                                             geneID</span></span>
<span><span class="co">## GO:0019882                                         CTSS/FCER1G/B2M/CTSL/CD74/CTSD/CTSH/FGL2/HLA-A/HLA-C/HLA-DMA/HLA-DPA1/HLA-DPB1/HLA-DQA1/HLA-DQA2/HLA-DQB1/HLA-DRA/HLA-DRB1/HLA-DRB5/HLA-E/IFI30/LILRB2/PSMB8/PSME1/PYCARD/RAB10/RAB32/SLC11A1/WAS/HLA-B/HLA-DMB/CALR/PDIA3/TAP1</span></span>
<span><span class="co">## GO:0022904                                                          BID/COX4I1/ISCU/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB4/NDUFB9/NDUFC2/NDUFS5/NDUFS7/PARK7/SOD2/UQCR10/UQCRFS1/UQCRH/SDHB/NDUFA3/NDUFAB1/SDHC/NDUFV2/UQCRQ/DGUOK/SCO2/GHITM/NDUFB7/SDHD/NDUFA2/NDUFS8/NDUFS2</span></span>
<span><span class="co">## GO:0046034 TSPO/BID/COX4I1/FIS1/HSPA8/ISCU/LDHA/NDUFA1/NDUFA4/NDUFB1/NDUFB10/NDUFB2/NDUFB4/NDUFB9/NDUFC2/NDUFS5/NDUFS7/PARK7/PGAM1/PGK1/PKM/RHOA/TPI1/UQCR10/UQCRC2/UQCRFS1/UQCRH/NDUFA3/NDUFAB1/TGFB1/SDHC/NDUFV2/UQCRQ/DGUOK/GHITM/NDUFB7/SDHD/UQCRC1/NCOR1/NDUFA2/NDUFS8/NDUFS2</span></span>
<span><span class="co">## GO:0033108                                                                                        COX14/NDUFA1/NDUFA11/NDUFA13/NDUFB1/NDUFB10/NDUFB11/NDUFB2/NDUFB4/NDUFB9/NDUFC2/NDUFS5/NDUFS7/UQCRFS1/NDUFA3/NDUFAB1/COX17/SCO2/NDUFB7/NDUFAF3/NDUFA2/NDUFS8/OXA1L/NDUFS2/PET100</span></span>
<span><span class="co">##            Count</span></span>
<span><span class="co">## GO:0019882    34</span></span>
<span><span class="co">## GO:0022904    32</span></span>
<span><span class="co">## GO:0046034    42</span></span>
<span><span class="co">## GO:0033108    25</span></span></code></pre>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Plot the top 4 terms per community</span></span>
<span><span class="va">dot_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_go_terms.html">plot_go_terms</a></span><span class="op">(</span><span class="va">enrich_results</span>, top_n_terms <span class="op">=</span> <span class="fl">5</span>, ontology <span class="op">=</span> <span class="st">"BP"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">dot_plot</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-8-1.png" alt="Dot plot of enriched terms" width="672"></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">## Save Results</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="hub-gene-changes-across-pseudotime">Hub Gene Changes Across Pseudotime<a class="anchor" aria-label="anchor" href="#hub-gene-changes-across-pseudotime"></a>
</h2>
<p>Identify top hub genes in each module and tabulate their presence
across bins.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hub_changes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">comm_results</span>, <span class="kw">function</span><span class="op">(</span><span class="va">res</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">bin</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">bin</span></span>
<span>  <span class="va">hub_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/seq.html" class="external-link">seq_along</a></span><span class="op">(</span><span class="va">res</span><span class="op">$</span><span class="va">communities</span><span class="op">)</span>, <span class="kw">function</span><span class="op">(</span><span class="va">i</span><span class="op">)</span> <span class="op">{</span></span>
<span>    <span class="va">genes</span> <span class="op">&lt;-</span> <span class="va">res</span><span class="op">$</span><span class="va">communities</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span></span>
<span>    <span class="va">subgraph</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/subgraph.html" class="external-link">induced_subgraph</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">[[</span><span class="va">bin</span><span class="op">]</span><span class="op">]</span>, <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/V.html" class="external-link">V</a></span><span class="op">(</span><span class="va">pt_results</span><span class="op">[[</span><span class="va">bin</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">[</span><span class="va">name</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">genes</span><span class="op">]</span><span class="op">)</span></span>
<span>    <span class="kw">if</span> <span class="op">(</span><span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/gorder.html" class="external-link">vcount</a></span><span class="op">(</span><span class="va">subgraph</span><span class="op">)</span> <span class="op">==</span> <span class="fl">0</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="cn">NULL</span><span class="op">)</span></span>
<span>    <span class="va">deg</span> <span class="op">&lt;-</span> <span class="fu">igraph</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/igraph/man/degree.html" class="external-link">degree</a></span><span class="op">(</span><span class="va">subgraph</span><span class="op">)</span></span>
<span>    <span class="va">top_hubs</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/sort.html" class="external-link">sort</a></span><span class="op">(</span><span class="va">deg</span>, decreasing <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/Extremes.html" class="external-link">min</a></span><span class="op">(</span><span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">deg</span><span class="op">)</span><span class="op">)</span><span class="op">]</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>Bin <span class="op">=</span> <span class="va">bin</span>, Community <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="st">"Group"</span>, <span class="va">i</span><span class="op">)</span>, Hub_Gene <span class="op">=</span> <span class="va">top_hubs</span>, stringsAsFactors <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span>  <span class="op">}</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/do.call.html" class="external-link">do.call</a></span><span class="op">(</span><span class="va">rbind</span>, <span class="va">hub_data</span><span class="op">)</span></span>
<span><span class="op">}</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Visualize hub gene presence across bins</span></span>
<span><span class="va">hub_presence</span> <span class="op">&lt;-</span> <span class="va">hub_changes</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">Hub_Gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>Bins_Present <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">Bin</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>, Communities <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">Community</span><span class="op">)</span>, collapse <span class="op">=</span> <span class="st">", "</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">"drop"</span><span class="op">)</span></span>
<span><span class="va">ft_hub</span> <span class="op">&lt;-</span> <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/flextable.html" class="external-link">flextable</a></span><span class="op">(</span><span class="va">hub_presence</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/autofit.html" class="external-link">autofit</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/set_header_labels.html" class="external-link">set_header_labels</a></span><span class="op">(</span>Hub_Gene <span class="op">=</span> <span class="st">"Hub Gene"</span>, Bins_Present <span class="op">=</span> <span class="st">"Bins Present"</span>, Communities <span class="op">=</span> <span class="st">"Communities"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/bg.html" class="external-link">bg</a></span><span class="op">(</span>part <span class="op">=</span> <span class="st">"header"</span>, bg <span class="op">=</span> <span class="st">"gray"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">flextable</span><span class="fu">::</span><span class="fu"><a href="https://davidgohel.github.io/flextable/reference/theme_vanilla.html" class="external-link">theme_vanilla</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">ft_hub</span></span></code></pre></div>
<div class="tabwid">
<style>.cl-547de958{}.cl-5471d604{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-5471d618{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-547a1f62{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-547a3b14{width:1.073in;background-color:rgba(190, 190, 190, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b28{width:2.087in;background-color:rgba(190, 190, 190, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b29{width:1.611in;background-color:rgba(190, 190, 190, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b32{width:1.073in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b33{width:2.087in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b3c{width:1.611in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b3d{width:1.073in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b46{width:2.087in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b50{width:1.611in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b51{width:1.073in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b5a{width:2.087in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-547a3b5b{width:1.611in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="table cl-547de958">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-547a3b14"><p class="cl-547a1f62"><span class="cl-5471d604">Hub Gene</span></p></th>
<th class="cl-547a3b28"><p class="cl-547a1f62"><span class="cl-5471d604">Bins Present</span></p></th>
<th class="cl-547a3b29"><p class="cl-547a1f62"><span class="cl-5471d604">Communities</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b32"><p class="cl-547a1f62"><span class="cl-5471d618">CAPZB</span></p></td>
<td class="cl-547a3b33"><p class="cl-547a1f62"><span class="cl-5471d618">PT1</span></p></td>
<td class="cl-547a3b3c"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">HIGD2A</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT3, PT3</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">HLA-DPA1</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT3, PT3</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">HLA-DRB5</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT1, PT2</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">ITGB2</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT2</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">LSP1</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT3</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">MYL12B</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT3</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">NEDD8</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT2, PT2</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">NPM1</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT3, PT3</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">PSMA7</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT2, PT2</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">S100A8</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT1, PT2, PT2, PT3, PT3</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1, Group 2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">S100A9</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT1</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">SLC25A3</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT1</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b3d"><p class="cl-547a1f62"><span class="cl-5471d618">TMA7</span></p></td>
<td class="cl-547a3b46"><p class="cl-547a1f62"><span class="cl-5471d618">PT2</span></p></td>
<td class="cl-547a3b50"><p class="cl-547a1f62"><span class="cl-5471d618">Group 1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-547a3b51"><p class="cl-547a1f62"><span class="cl-5471d618">UBB</span></p></td>
<td class="cl-547a3b5a"><p class="cl-547a1f62"><span class="cl-5471d618">PT2</span></p></td>
<td class="cl-547a3b5b"><p class="cl-547a1f62"><span class="cl-5471d618">Group 2</span></p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="hub-genes">Hub Genes<a class="anchor" aria-label="anchor" href="#hub-genes"></a>
</h2>
<p>Identify top hub genes for communities in each pseudotime bin.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ft</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/summarize_hub_genes_across_bins.html">summarize_hub_genes_across_bins</a></span><span class="op">(</span><span class="va">comm_results</span>, <span class="va">pt_results</span>, top_n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="va">ft</span></span></code></pre></div>
<div class="tabwid">
<style>.cl-54a48fae{}.cl-549f602e{font-family:'DejaVu Sans';font-size:11pt;font-weight:bold;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-549f6042{font-family:'DejaVu Sans';font-size:11pt;font-weight:normal;font-style:normal;text-decoration:none;color:rgba(0, 0, 0, 1.00);background-color:transparent;}.cl-54a1a208{margin:0;text-align:left;border-bottom: 0 solid rgba(0, 0, 0, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);padding-bottom:5pt;padding-top:5pt;padding-left:5pt;padding-right:5pt;line-height: 1;background-color:transparent;}.cl-54a1b05e{width:0.568in;background-color:rgba(190, 190, 190, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b068{width:0.899in;background-color:rgba(190, 190, 190, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b072{width:7.313in;background-color:rgba(190, 190, 190, 1.00);vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 1.5pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b073{width:0.568in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b07c{width:0.899in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b086{width:7.313in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0 solid rgba(0, 0, 0, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b090{width:0.568in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b091{width:0.899in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b092{width:7.313in;background-color:transparent;vertical-align: middle;border-bottom: 0.75pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b09a{width:0.568in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b0a4{width:0.899in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}.cl-54a1b0ae{width:7.313in;background-color:transparent;vertical-align: middle;border-bottom: 1.5pt solid rgba(102, 102, 102, 1.00);border-top: 0.75pt solid rgba(102, 102, 102, 1.00);border-left: 0 solid rgba(0, 0, 0, 1.00);border-right: 0 solid rgba(0, 0, 0, 1.00);margin-bottom:0;margin-top:0;margin-left:0;margin-right:0;}</style>
<table data-quarto-disable-processing="true" class="table cl-54a48fae">
<thead><tr style="overflow-wrap:break-word;">
<th class="cl-54a1b05e"><p class="cl-54a1a208"><span class="cl-549f602e">Bin</span></p></th>
<th class="cl-54a1b068"><p class="cl-54a1a208"><span class="cl-549f602e">Group</span></p></th>
<th class="cl-54a1b072"><p class="cl-54a1a208"><span class="cl-549f602e">Top Hub Genes</span></p></th>
</tr></thead>
<tbody>
<tr style="overflow-wrap:break-word;">
<td class="cl-54a1b073"><p class="cl-54a1a208"><span class="cl-549f6042">PT1</span></p></td>
<td class="cl-54a1b07c"><p class="cl-54a1a208"><span class="cl-549f6042">Group 1</span></p></td>
<td class="cl-54a1b086"><p class="cl-54a1a208"><span class="cl-549f6042">S100A8, HLA-DRB5, S100A9, CAPZB, SLC25A3, RHOG, UBE2D3, NDUFA11, ATP6V0E1, BST2</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-54a1b090"><p class="cl-54a1a208"><span class="cl-549f6042">PT2</span></p></td>
<td class="cl-54a1b091"><p class="cl-54a1a208"><span class="cl-549f6042">Group 1</span></p></td>
<td class="cl-54a1b092"><p class="cl-54a1a208"><span class="cl-549f6042">S100A8, PSMA7, NEDD8, TMA7, ITGB2, UBB, SSR4, LGALS9, HLA-DRB5, CSTB</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-54a1b091"><p class="cl-54a1a208"><span class="cl-549f6042">Group 2</span></p></td>
<td class="cl-54a1b092"><p class="cl-54a1a208"><span class="cl-549f6042">S100A8, PSMA7, NEDD8, UBB, HLA-DRB5, ITGB2, TMA7, SSR4, LGALS9, HLA-DPA1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td rowspan="2" class="cl-54a1b09a"><p class="cl-54a1a208"><span class="cl-549f6042">PT3</span></p></td>
<td class="cl-54a1b091"><p class="cl-54a1a208"><span class="cl-549f6042">Group 1</span></p></td>
<td class="cl-54a1b092"><p class="cl-54a1a208"><span class="cl-549f6042">HLA-DPA1, NPM1, S100A8, HIGD2A, LSP1, MYL12B, ATP5D, C11orf31, CIRBP, SPI1</span></p></td>
</tr>
<tr style="overflow-wrap:break-word;">
<td class="cl-54a1b0a4"><p class="cl-54a1a208"><span class="cl-549f6042">Group 2</span></p></td>
<td class="cl-54a1b0ae"><p class="cl-54a1a208"><span class="cl-549f6042">HLA-DPA1, NPM1, S100A8, HIGD2A, MYL12B, SPI1, LSP1, S100A9, CD74, SLC25A5</span></p></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="edge-gainloss-analysis">Edge Gain/Loss Analysis<a class="anchor" aria-label="anchor" href="#edge-gainloss-analysis"></a>
</h2>
<p>Compare stable, gained, and lost edges between consecutive pseudotime
bins.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_edge_gain_loss.html">plot_edge_gain_loss</a></span><span class="op">(</span><span class="va">pt_results</span>, cluster_id <span class="op">=</span> <span class="st">"C1"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Edges in PT1 : 211806 </span></span>
<span><span class="co">## Edges in PT2 : 211825 </span></span>
<span><span class="co">## Stable edges ( PT1 to PT2 ): 87964 </span></span>
<span><span class="co">## Gained edges ( PT1 to PT2 ): 123861 </span></span>
<span><span class="co">## Lost edges ( PT1 to PT2 ): 123842 </span></span>
<span><span class="co">## Edges in PT2 : 211825 </span></span>
<span><span class="co">## Edges in PT3 : 211818 </span></span>
<span><span class="co">## Stable edges ( PT2 to PT3 ): 91598 </span></span>
<span><span class="co">## Gained edges ( PT2 to PT3 ): 120220 </span></span>
<span><span class="co">## Lost edges ( PT2 to PT3 ): 120227</span></span></code></pre>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_gl</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">plot</span>  <span class="co"># Correct assignment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_gl</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-11-1.png" alt="bar plot of edges changes between bins" width="768"></p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Inspect specific edges (e.g., for PT1 to PT2)</span></span>
<span><span class="va">pt1_to_pt2_edges</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">edge_details</span><span class="op">[[</span><span class="st">"PT1 to PT2"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Gained edges (PT1 to PT2):\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Gained edges (PT1 to PT2):</span></span></code></pre>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">gained</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">gained</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">gained</span>, <span class="st">"gained_edges_PT1_to_PT2.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"No gained edges found.\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">##    edge_changes$gained</span></span>
<span><span class="co">## 1          CCL3_S100A8</span></span>
<span><span class="co">## 2        MT-CO2_S100A8</span></span>
<span><span class="co">## 3         S100A8_ZFP36</span></span>
<span><span class="co">## 4       MT-ATP6_S100A8</span></span>
<span><span class="co">## 5          LY6E_S100A8</span></span>
<span><span class="co">## 6     LINC00936_S100A8</span></span>
<span><span class="co">## 7      C1orf162_S100A8</span></span>
<span><span class="co">## 8         ANXA2_S100A8</span></span>
<span><span class="co">## 9       NDUFA13_S100A8</span></span>
<span><span class="co">## 10         CD63_S100A8</span></span></code></pre>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Lost edges (PT1 to PT2):\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Lost edges (PT1 to PT2):</span></span></code></pre>
<div class="sourceCode" id="cb27"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">lost</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">lost</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">lost</span>, <span class="st">"lost_edges_PT1_to_PT2.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"No lost edges found.\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">##    edge_changes$lost</span></span>
<span><span class="co">## 1      S100A8_STXBP2</span></span>
<span><span class="co">## 2        GPX4_S100A8</span></span>
<span><span class="co">## 3        GMFG_S100A8</span></span>
<span><span class="co">## 4       S100A8_SF3B5</span></span>
<span><span class="co">## 5        EMP3_S100A8</span></span>
<span><span class="co">## 6       CNPY3_S100A8</span></span>
<span><span class="co">## 7        CD37_S100A8</span></span>
<span><span class="co">## 8     S100A8_WDR83OS</span></span>
<span><span class="co">## 9       S100A8_UQCRH</span></span>
<span><span class="co">## 10    ANAPC16_S100A8</span></span></code></pre>
<div class="sourceCode" id="cb29"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Stable edges (PT1 to PT2):\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Stable edges (PT1 to PT2):</span></span></code></pre>
<div class="sourceCode" id="cb31"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">stable</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">stable</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">pt1_to_pt2_edges</span><span class="op">$</span><span class="va">stable</span>, <span class="st">"stable_edges_PT1_to_PT2.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"No stable edges found.\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">##    edge_changes$stable</span></span>
<span><span class="co">## 1        S100A8_S100A9</span></span>
<span><span class="co">## 2        FCGR3A_S100A8</span></span>
<span><span class="co">## 3        LGALS2_S100A8</span></span>
<span><span class="co">## 4      HLA-DPB1_S100A8</span></span>
<span><span class="co">## 5        IFITM2_S100A8</span></span>
<span><span class="co">## 6        IFITM3_S100A8</span></span>
<span><span class="co">## 7      HLA-DPA1_S100A8</span></span>
<span><span class="co">## 8          FCN1_S100A8</span></span>
<span><span class="co">## 9          GPX1_S100A8</span></span>
<span><span class="co">## 10       NFKBIA_S100A8</span></span></code></pre>
<div class="sourceCode" id="cb33"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Repeat for PT2 to PT3</span></span>
<span><span class="va">pt2_to_pt3_edges</span> <span class="op">&lt;-</span> <span class="va">result</span><span class="op">$</span><span class="va">edge_details</span><span class="op">[[</span><span class="st">"PT2 to PT3"</span><span class="op">]</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"Gained edges (PT2 to PT3):\n"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## Gained edges (PT2 to PT3):</span></span></code></pre>
<div class="sourceCode" id="cb35"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NULL.html" class="external-link">is.null</a></span><span class="op">(</span><span class="va">pt2_to_pt3_edges</span><span class="op">$</span><span class="va">gained</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pt2_to_pt3_edges</span><span class="op">$</span><span class="va">gained</span>, <span class="fl">10</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">pt2_to_pt3_edges</span><span class="op">$</span><span class="va">gained</span>, <span class="st">"gained_edges_PT2_to_PT3.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span><span class="op">}</span> <span class="kw">else</span> <span class="op">{</span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cat.html" class="external-link">cat</a></span><span class="op">(</span><span class="st">"No gained edges found.\n"</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre></div>
<pre><code><span><span class="co">##    edge_changes$gained</span></span>
<span><span class="co">## 1        MT-ND2_S100A8</span></span>
<span><span class="co">## 2        S100A8_STXBP2</span></span>
<span><span class="co">## 3          EMP3_S100A8</span></span>
<span><span class="co">## 4          GPX4_S100A8</span></span>
<span><span class="co">## 5         ALDH2_S100A8</span></span>
<span><span class="co">## 6        EIF4A1_S100A8</span></span>
<span><span class="co">## 7         CLIC1_S100A8</span></span>
<span><span class="co">## 8          GMFG_S100A8</span></span>
<span><span class="co">## 9         S100A8_SF3B5</span></span>
<span><span class="co">## 10        CNPY3_S100A8</span></span></code></pre>
<p>To further explore dynamic gene interactions, we used
<code><a href="../reference/plot_edge_gain_loss.html">plot_edge_gain_loss()</a></code> to quantify and visualize stable,
gained, and lost edges between consecutive pseudotime bins. The function
returns both a summary plot and detailed edge lists for each transition.
For example, from <strong>PT1→PT2</strong>, over 120K edges were gained
or lost, and ~87K remained stable. Exported edge tables (e.g.,
‘gained_edges_PT1_to_PT2.csv’) can be used for downstream analysis of
context-specific co-regulation.</p>
</div>
<div class="section level2">
<h2 id="gene-pseudotime-correlation-density">Gene-Pseudotime Correlation Density<a class="anchor" aria-label="anchor" href="#gene-pseudotime-correlation-density"></a>
</h2>
<p>Plot distribution of gene expression–pseudotime correlations to gauge
overall trends.</p>
<div class="sourceCode" id="cb37"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gc_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/gene_pt_correlation.html">gene_pt_correlation</a></span><span class="op">(</span></span>
<span>  <span class="va">seurat_obj</span><span class="op">[</span>, <span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">RNA_snn_res.0.1</span> <span class="op">==</span> <span class="st">"1"</span><span class="op">]</span>,</span>
<span>  pseudotime_col <span class="op">=</span> <span class="st">"pseudotime"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, layer <span class="op">=</span> <span class="st">"data"</span>, method <span class="op">=</span> <span class="st">"spearman"</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Cluster <span class="op">=</span> <span class="st">"C1"</span><span class="op">)</span></span>
<span><span class="va">p_gc</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">gc_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Cor</span>, color <span class="op">=</span> <span class="va">Cluster</span>, fill <span class="op">=</span> <span class="va">Cluster</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_density.html" class="external-link">geom_density</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.3</span>, color <span class="op">=</span> <span class="st">"#00B6EB"</span>, fill <span class="op">=</span> <span class="st">"#00B6EB"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Gene-Pseudotime Correlation Density"</span>, x <span class="op">=</span> <span class="st">"Spearman ρ"</span>, y <span class="op">=</span> <span class="st">"Density"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_gc</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-12-1.png" alt="Distriution of gene correlations with pseudotime" width="768"></p>
</div>
<div class="section level2">
<h2 id="top-10-genes-by-correlation">Top ±10 Genes by Correlation<a class="anchor" aria-label="anchor" href="#top-10-genes-by-correlation"></a>
</h2>
<p>Plot top 10 positively and negatively correlated genes.</p>
<div class="sourceCode" id="cb38"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top_genes_df</span> <span class="op">&lt;-</span> <span class="va">gc_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Cor</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_max</a></span><span class="op">(</span><span class="va">Cor</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="st">"Positive"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/bind_rows.html" class="external-link">bind_rows</a></span><span class="op">(</span></span>
<span>    <span class="va">gc_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">Cor</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_min</a></span><span class="op">(</span><span class="va">Cor</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Direction <span class="op">=</span> <span class="st">"Negative"</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>Gene <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html" class="external-link">factor</a></span><span class="op">(</span><span class="va">Gene</span>, levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">p_top_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">top_genes_df</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">Cor</span>, y <span class="op">=</span> <span class="va">Gene</span>, fill <span class="op">=</span> <span class="va">Direction</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_col</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_grid.html" class="external-link">facet_grid</a></span><span class="op">(</span><span class="va">Direction</span> <span class="op">~</span> <span class="va">Cluster</span>, scales <span class="op">=</span> <span class="st">"free_y"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Top ±10 Genes by Pseudotime Correlation"</span>, x <span class="op">=</span> <span class="st">"Spearman ρ"</span>, y <span class="op">=</span> <span class="st">""</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_top_genes</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-13-1.png" alt="Bar plot of genes correlated with pseudotime" width="576"></p>
</div>
<div class="section level2">
<h2 id="heatmap-of-top-10-genes">Heatmap of Top 10 Genes<a class="anchor" aria-label="anchor" href="#heatmap-of-top-10-genes"></a>
</h2>
<p>Heatmap of top 10 genes by pseudotime correlation.</p>
<div class="sourceCode" id="cb39"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">top10</span> <span class="op">&lt;-</span> <span class="va">gc_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/slice.html" class="external-link">slice_max</a></span><span class="op">(</span><span class="va">Cor</span>, n <span class="op">=</span> <span class="fl">10</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/pull.html" class="external-link">pull</a></span><span class="op">(</span><span class="va">Gene</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="va">valid_cells</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/row_colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">)</span><span class="op">[</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">RNA_snn_res.0.1</span> <span class="op">==</span> <span class="st">"1"</span> <span class="op">&amp;</span> <span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">$</span><span class="va">pseudotime</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">cells</span> <span class="op">&lt;-</span> <span class="va">valid_cells</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/order.html" class="external-link">order</a></span><span class="op">(</span><span class="va">seurat_obj</span><span class="op">@</span><span class="va">meta.data</span><span class="op">[</span><span class="va">valid_cells</span>, <span class="st">"pseudotime"</span><span class="op">]</span><span class="op">)</span><span class="op">]</span></span>
<span><span class="va">expr</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/SeuratObject/man/AssayData.html" class="external-link">GetAssayData</a></span><span class="op">(</span><span class="va">seurat_obj</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>, slot <span class="op">=</span> <span class="st">"data"</span><span class="op">)</span></span>
<span><span class="va">present</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">intersect</a></span><span class="op">(</span><span class="va">top10</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">missing</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">top10</span>, <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">expr</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">expr</span><span class="op">[</span><span class="va">present</span>, <span class="va">cells</span>, drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="kw">if</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">missing</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">mat2</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/matrix.html" class="external-link">matrix</a></span><span class="op">(</span><span class="cn">NA</span>, nrow <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">missing</span><span class="op">)</span>, ncol <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">cells</span><span class="op">)</span>, dimnames <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span><span class="va">missing</span>, <span class="va">cells</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="va">mat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">mat</span>, <span class="va">mat2</span><span class="op">)</span></span>
<span><span class="op">}</span></span>
<span><span class="va">mat</span> <span class="op">&lt;-</span> <span class="va">mat</span><span class="op">[</span><span class="va">top10</span>, , drop <span class="op">=</span> <span class="cn">FALSE</span><span class="op">]</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/pheatmap/man/pheatmap.html" class="external-link">pheatmap</a></span><span class="op">(</span><span class="va">mat</span>, cluster_rows <span class="op">=</span> <span class="cn">FALSE</span>, cluster_cols <span class="op">=</span> <span class="cn">FALSE</span>, show_colnames <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>         main <span class="op">=</span> <span class="st">"Top 10 Genes by Pseudotime Correlation"</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-14-1.png" alt="Heatmap of top 10 genes correlated with pseudotime" width="768"></p>
</div>
<div class="section level2">
<h2 id="module-score-trajectories">Module-Score Trajectories<a class="anchor" aria-label="anchor" href="#module-score-trajectories"></a>
</h2>
<p>Compute and plot module scores for PT2 communities along
pseudotime.</p>
<div class="sourceCode" id="cb40"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Define custom modules to plot</span></span>
<span><span class="va">custom_mods</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  M1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'S100A8'</span>, <span class="st">'UBB'</span>, <span class="st">'SSR4'</span><span class="op">)</span>,</span>
<span>  M2 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'LGALS9'</span>, <span class="st">'HLA-DPA1'</span><span class="op">)</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Plot trajectories for custom modules in PT2</span></span>
<span><span class="va">p_modtraj</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_module_trajectories.html">plot_module_trajectories</a></span><span class="op">(</span></span>
<span>  <span class="va">comm_results</span>, <span class="va">seurat_obj</span>, bin_index <span class="op">=</span> <span class="fl">2</span>, </span>
<span>  custom_modules <span class="op">=</span> <span class="va">custom_mods</span>,</span>
<span>  cluster_field <span class="op">=</span> <span class="st">"RNA_snn_res.0.1"</span>, cluster_id <span class="op">=</span> <span class="st">"1"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_modtraj</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-15-1.png" alt="Module trajectory plot" width="768"></p>
</div>
<div class="section level2">
<h2 id="edge-pseudotime-correlation-density">Edge-Pseudotime Correlation Density<a class="anchor" aria-label="anchor" href="#edge-pseudotime-correlation-density"></a>
</h2>
<p>Compute and plot the density of edge-pseudotime correlations for a
specified pseudotime bin. We evaluated how gene-gene interactions change
with pseudotime by computing Spearman correlations between edge
expression and pseudotime in PT2. Using
<code><a href="../reference/plot_edge_correlations.html">plot_edge_correlations()</a></code>, we observed that edge
correlations are centered around zero across all groups. While Community
1 and Community 2 show similar distributions, subtle differences suggest
distinct temporal dynamics within and across gene modules.</p>
<div class="sourceCode" id="cb41"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p_ec</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/plot_edge_correlations.html">plot_edge_correlations</a></span><span class="op">(</span></span>
<span>  <span class="va">comm_results</span>, <span class="va">pt_results</span>, <span class="va">seurat_obj</span>, bin_name <span class="op">=</span> <span class="st">"PT2"</span>,</span>
<span>  cluster_field <span class="op">=</span> <span class="st">"RNA_snn_res.0.1"</span>, cluster_id <span class="op">=</span> <span class="st">"1"</span>, assay <span class="op">=</span> <span class="st">"RNA"</span>,</span>
<span>  layer <span class="op">=</span> <span class="st">"data"</span>, pseudotime_col <span class="op">=</span> <span class="st">"pseudotime"</span>, method <span class="op">=</span> <span class="st">"spearman"</span>,</span>
<span>  edge_threshold <span class="op">=</span> <span class="fl">0.95</span>, n_cores <span class="op">=</span> <span class="fl">36</span></span>
<span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">p_ec</span><span class="op">)</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-16-1.png" alt="Edge correlation histogram" width="768"></p>
</div>
<div class="section level2">
<h2 id="combined-summary-plots">Combined Summary Plots<a class="anchor" aria-label="anchor" href="#combined-summary-plots"></a>
</h2>
<p>Arrange key plots into a composite figure for quick overview.</p>
<div class="sourceCode" id="cb42"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="op">(</span><span class="va">p_pt</span> <span class="op">|</span> <span class="va">p_gc</span><span class="op">)</span> <span class="op">/</span> <span class="op">(</span><span class="va">p_ec</span> <span class="op">|</span> <span class="va">p_gl</span><span class="op">)</span> <span class="op">/</span> <span class="va">p_modtraj</span></span></code></pre></div>
<p><img src="pseudotime_mi_network_files/figure-html/unnamed-chunk-17-1.png" alt="Combined plot" width="1152"></p>
</div>
<div class="section level2">
<h2 id="save-results">Save Results<a class="anchor" aria-label="anchor" href="#save-results"></a>
</h2>
<div class="sourceCode" id="cb43"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># saveRDS(pt_results, "pseudotime_results.rds")</span></span>
<span><span class="co"># saveRDS(comm_results, "pseudotime_communities.rds")</span></span>
<span><span class="co"># saveRDS(enrich_results, "enrich_results.rds")</span></span></code></pre></div>
<div class="sourceCode" id="cb44"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Red Hat Enterprise Linux 9.4 (Plow)</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: /app1/ebapps/arches/flat-avx2/software/FlexiBLAS/3.2.0-GCC-11.3.0/lib64/libflexiblas.so.3.2</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] DOSE_3.22.1           flextable_0.9.2       ggraph_2.0.5         </span></span>
<span><span class="co">##  [4] miEdgeR_0.1.0         pheatmap_1.0.12       ggVennDiagram_1.2.3  </span></span>
<span><span class="co">##  [7] ggpubr_0.4.0          org.Hs.eg.db_3.15.0   AnnotationDbi_1.58.0 </span></span>
<span><span class="co">## [10] IRanges_2.32.0        S4Vectors_0.36.1      Biobase_2.56.0       </span></span>
<span><span class="co">## [13] BiocGenerics_0.44.0   clusterProfiler_4.4.4 ggplot2_3.3.6        </span></span>
<span><span class="co">## [16] dplyr_1.1.4           igraph_1.3.2          SeuratObject_4.1.3   </span></span>
<span><span class="co">## [19] Seurat_4.3.0         </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rappdirs_0.3.3          scattermore_1.2         ragg_1.2.5             </span></span>
<span><span class="co">##   [4] tidyr_1.2.0             bit64_4.0.5             knitr_1.50             </span></span>
<span><span class="co">##   [7] irlba_2.3.5             data.table_1.14.2       KEGGREST_1.36.2        </span></span>
<span><span class="co">##  [10] RCurl_1.98-1.7          generics_0.1.3          cowplot_1.1.1          </span></span>
<span><span class="co">##  [13] RSQLite_2.2.14          shadowtext_0.1.2        RANN_2.6.1             </span></span>
<span><span class="co">##  [16] future_1.26.1           bit_4.0.4               enrichplot_1.16.2      </span></span>
<span><span class="co">##  [19] spatstat.data_3.0-0     xml2_1.3.3              httpuv_1.6.5           </span></span>
<span><span class="co">##  [22] viridis_0.6.2           xfun_0.52               jquerylib_0.1.4        </span></span>
<span><span class="co">##  [25] evaluate_0.15           promises_1.2.0.1        DBI_1.1.3              </span></span>
<span><span class="co">##  [28] htmlwidgets_1.5.4       spatstat.geom_3.0-3     purrr_1.0.4            </span></span>
<span><span class="co">##  [31] ellipsis_0.3.2          backports_1.4.1         fontLiberation_0.1.0   </span></span>
<span><span class="co">##  [34] fontBitstreamVera_0.1.1 deldir_1.0-6            vctrs_0.6.5            </span></span>
<span><span class="co">##  [37] ROCR_1.0-11             abind_1.4-5             cachem_1.0.6           </span></span>
<span><span class="co">##  [40] withr_2.5.0             ggforce_0.3.3           RVenn_1.1.0            </span></span>
<span><span class="co">##  [43] progressr_0.10.1        sctransform_0.4.1       treeio_1.20.2          </span></span>
<span><span class="co">##  [46] goftest_1.2-3           cluster_2.1.3           ape_5.6-2              </span></span>
<span><span class="co">##  [49] lazyeval_0.2.2          crayon_1.5.1            crul_1.2.0             </span></span>
<span><span class="co">##  [52] spatstat.explore_3.0-5  labeling_0.4.2          pkgconfig_2.0.3        </span></span>
<span><span class="co">##  [55] tweenr_1.0.2            GenomeInfoDb_1.32.2     nlme_3.1-158           </span></span>
<span><span class="co">##  [58] rlang_1.1.5             globals_0.15.0          lifecycle_1.0.4        </span></span>
<span><span class="co">##  [61] miniUI_0.1.1.1          downloader_0.4          fontquiver_0.2.1       </span></span>
<span><span class="co">##  [64] httpcode_0.3.0          rprojroot_2.0.3         polyclip_1.10-0        </span></span>
<span><span class="co">##  [67] matrixStats_0.62.0      lmtest_0.9-40           graph_1.74.0           </span></span>
<span><span class="co">##  [70] Matrix_1.5-3            aplot_0.1.9             carData_3.0-5          </span></span>
<span><span class="co">##  [73] zoo_1.8-10              ggridges_0.5.3          png_0.1-7              </span></span>
<span><span class="co">##  [76] viridisLite_0.4.0       bitops_1.0-7            KernSmooth_2.23-20     </span></span>
<span><span class="co">##  [79] Biostrings_2.64.0       blob_1.2.3              stringr_1.4.0          </span></span>
<span><span class="co">##  [82] qvalue_2.28.0           parallelly_1.32.0       spatstat.random_3.0-1  </span></span>
<span><span class="co">##  [85] rstatix_0.7.0           gridGraphics_0.5-1      ggsignif_0.6.3         </span></span>
<span><span class="co">##  [88] reactome.db_1.81.0      scales_1.2.0            memoise_2.0.1          </span></span>
<span><span class="co">##  [91] graphite_1.42.0         magrittr_2.0.3          plyr_1.8.7             </span></span>
<span><span class="co">##  [94] ica_1.0-2               zlibbioc_1.42.0         compiler_4.2.1         </span></span>
<span><span class="co">##  [97] scatterpie_0.1.8        RColorBrewer_1.1-3      fitdistrplus_1.1-8     </span></span>
<span><span class="co">## [100] cli_3.6.4               XVector_0.36.0          listenv_0.8.0          </span></span>
<span><span class="co">## [103] patchwork_1.1.1         pbapply_1.5-0           mgcv_1.8-40            </span></span>
<span><span class="co">## [106] MASS_7.3-57             tidyselect_1.2.1        stringi_1.7.6          </span></span>
<span><span class="co">## [109] textshaping_0.3.6       yaml_2.3.10             GOSemSim_2.22.0        </span></span>
<span><span class="co">## [112] askpass_1.1             ggrepel_0.9.1           grid_4.2.1             </span></span>
<span><span class="co">## [115] sass_0.4.10             fastmatch_1.1-3         tools_4.2.1            </span></span>
<span><span class="co">## [118] future.apply_1.9.0      parallel_4.2.1          rstudioapi_0.13        </span></span>
<span><span class="co">## [121] uuid_1.1-0              gridExtra_2.3           farver_2.1.0           </span></span>
<span><span class="co">## [124] Rtsne_0.16              digest_0.6.29           shiny_1.7.1            </span></span>
<span><span class="co">## [127] gfonts_0.2.0            Rcpp_1.0.8.3            car_3.1-0              </span></span>
<span><span class="co">## [130] broom_0.8.0             infotheo_1.2.0.1        later_1.3.0            </span></span>
<span><span class="co">## [133] RcppAnnoy_0.0.20        httr_1.4.3              gdtools_0.3.3          </span></span>
<span><span class="co">## [136] colorspace_2.0-3        fs_1.5.2                tensor_1.5             </span></span>
<span><span class="co">## [139] reticulate_1.25         splines_4.2.1           uwot_0.1.14            </span></span>
<span><span class="co">## [142] yulab.utils_0.2.0       tidytree_0.4.1          spatstat.utils_3.1-3   </span></span>
<span><span class="co">## [145] pkgdown_2.1.1           graphlayouts_0.8.0      sp_1.5-1               </span></span>
<span><span class="co">## [148] ggplotify_0.1.0         plotly_4.10.0           systemfonts_1.0.4      </span></span>
<span><span class="co">## [151] xtable_1.8-4            jsonlite_1.8.0          ggtree_3.4.4           </span></span>
<span><span class="co">## [154] tidygraph_1.2.1         ggfun_0.1.8             R6_2.5.1               </span></span>
<span><span class="co">## [157] pillar_1.10.1           htmltools_0.5.8.1       mime_0.12              </span></span>
<span><span class="co">## [160] glue_1.6.2              fastmap_1.2.0           BiocParallel_1.32.5    </span></span>
<span><span class="co">## [163] codetools_0.2-18        fgsea_1.22.0            lattice_0.20-45        </span></span>
<span><span class="co">## [166] bslib_0.9.0             spatstat.sparse_3.0-0   tibble_3.2.1           </span></span>
<span><span class="co">## [169] curl_6.2.2              leiden_0.4.2            ReactomePA_1.40.0      </span></span>
<span><span class="co">## [172] officer_0.6.2           zip_2.2.0               GO.db_3.15.0           </span></span>
<span><span class="co">## [175] openssl_2.0.2           survival_3.3-1          rmarkdown_2.29         </span></span>
<span><span class="co">## [178] desc_1.4.1              munsell_0.5.0           DO.db_2.9              </span></span>
<span><span class="co">## [181] GenomeInfoDbData_1.2.8  reshape2_1.4.4          gtable_0.3.0</span></span></code></pre>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Elhadi Iich.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
